spec/lib/ruby_smb/smb1/bit_field/file_status_flags_spec.rb

```diff
@@ -0,0 +1,26 @@
+RSpec.describe RubySMB::SMB1::BitField::FileStatusFlags do
+  subject(:options) { described_class.new }
+
+  it { is_expected.to respond_to :reparse_tag }
+  it { is_expected.to respond_to :no_substreams }
+  it { is_expected.to respond_to :no_eas }
+
+  it 'is little endian' do
+    expect(described_class.fields.instance_variable_get(:@hints)[:endian]).to eq :little
+  end
+
+  {
+    :reparse_tag    => 0x0004,
+    :no_substreams  => 0x0002,
+    :no_eas         => 0x0001
+  }.each do |field, bitmask|
+    describe "##{field.to_s}" do
+      it 'is a 1-bit flag' do
+        expect(options.send(field)).to be_a BinData::Bit1
+      end
```

Comment:
please do not ever do meta-programming in specs, this can cause serious problems down the line. Write each test out

Reply:
Oh ok, sorry, I didn't know about this. I will fix this, no problem.
Just out of curiosity, what kind of issue this could cause?

Reply:
one of the big ones is that you won't actually be able to tell which test scenario failed if one of them fails without a lot of digging, it turns into a mess really quickly

Reply:
That's a very good reason, thanks!

Reply:
Addressed in 37aad9e13a382c36c0e1a23a899e6c4f549a881f.